### ISSUE TITLE
Qutebrowser sometimes does not detect the correct keyboard layout

### ISSUE THREAD

**Issue by phansch** [qutebrowser/qutebrowser#2648]
My keyboard layout is `de(dvorak)` and it has been working flawlessly with qutebrowser most of the time. However, sometimes after starting qutebrowser, it thinks that I am using a QWERTY layout.
In other programs the keyboard layout is working fine.

I have to close qutebrowser in that case and run `setxkbmap 'de(dvorak)'` to make it work again.

This happened three times already but I unfortunately don't have any reproduction steps yet.
I will also add a :report the next time it happens. Is there anything else I can do to help find the cause of this?

Version info: https://gist.github.com/phansch/6a0b65eba6bbb0d49ce177ec98fca6c8

**Comment by The-Compiler** [comment]
Do you switch layouts from time to time? Does it only happen for web content, or in the commandline (`:foo`) as well? If it happens, does `:restart` do anything? Does the same happen in other Qt applications? What about other QtWebEngine applications like QupZilla (if you can get a 2.x version)?

**Comment by phansch** [comment]
> Do you switch layouts from time to time?

No, never.

> Does it only happen for web content, or in the commandline (:foo) as well? If it happens, does :restart do anything?

It also happens in the command line. For example I would have to enter `: f o o` completely with the QWERTY layout. I will try :restart the next time.

> Does the same happen in other Qt applications? What about other QtWebEngine applications like QupZilla (if you can get a 2.x version)?

Will also give it a try and let you know :+1:

**Comment by phansch** [comment]
I just created a report after the above thing happended. Using :restart did not fix the problem.

Using `setxkbmap -print -verbose 10` now shows me that the keyboard layout is indeed `us`.

However, qutebrowser seems to be the only affected application by that. I unfortunately didn't have time to set up QupZilla 2. I have hipchat4 installed which also seems to use QtWebEngine, and the layout is still the one from my `.xinitrc`, not `us`.

So something somehow resets the keyboard layout to `us`. I don't know if that's qutebrowser or something else. I will try to log when the keyboard layout is changed to `us` and maybe I can rule out qutebrowser :+1:

**Comment by phansch** [comment]
Ok, I just found out why the keyboard layout is reset: A few seconds after the machine wakes up from `pm-suspend` the layout gets set to `us`. That is even without qutebrowser running.

I recently installed the `xfce4-settings` configuration stuff so I guess there is some configuration that I need to update to set up the correct keyboard layout, apart from the `.xinitrc`.

I guess qutebrowser does everything correctly in detecting the `us` keyboard layout and this is not a qutebrowser bug :+1:
